Assets/Scripts/Achievement/AchievementController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using BTScriptableObject;
+using System;
+
+namespace AchievementM
+{
+    public class AchievementController
+    {
+        private Achievement achievementHolder = new Achievement();
+
+        public Achievement AchievementHolder { get { return achievementHolder; } }
+
+        private AchievementScriptable achievementScriptable;
+        private AchievementType achievementType;
+
+        public AchievementController(AchievementScriptable achievementScriptable, AchievementType achievementType)
+        {
+            this.achievementScriptable = achievementScriptable;
+            this.achievementType = achievementType;
+            SetAchievementData();
+        }
+
+        private void SetAchievementData()
+        {
+            foreach (Achievement item in achievementScriptable.achievements)
+            {
+                if (item.achievementType == achievementType)
+                {
+                    achievementHolder.AchievementName = item.AchievementName;
+                    achievementHolder.achievementType = item.achievementType;
+                    int r = item.achievementInfo.Count;
```

Comment:
no idea what r is and why you need it. if(item.achievementInfo.Count > 0) is much more readable